pickhardtpayments/Attempt.py

```diff
@@ -0,0 +1,136 @@
+    :type amount: int
+    """
+
+    def __init__(self, path: list[UncertaintyChannel], amount: int = 0):
+        """Constructor method
+        """
+        if amount >= 0:
+            self._amount = amount
+        else:
+            raise ValueError("amount for payment attempts needs to be positive")
+
+        i = 1
+        valid_path = True
+        while i < len(path):
+            valid_path = valid_path and (path[i - 1].dest == path[i].src)
+            i += 1
+        if valid_path:
+            self._path = path
+
+        channel: UncertaintyChannel
+        self._routing_fee = 0
+        self._probability = 1
+        for channel in path:
+            self._routing_fee += channel.routing_cost_msat(amount)
+            self._probability *= channel.success_probability(amount)
+            # When Attempt is created, all amounts are set inflight. Needs to be updated with AttemptStatus change!
+            # This is to correctly compute conditional probabilities of non-disjoint paths in the same set of paths
+            # channel.in_flight(amount)
+            channel.allocate_amount(amount)
+        self._status = AttemptStatus.PLANNED
+
+    def __str__(self):
+        description = "Path with {} channels to deliver {} sats and status {}.".format(len(self._path),
+                                                                                       self._amount, self._status.name)
+        if self._routing_fee and self._routing_fee > 0:
+            description += "\nsuccess probability of {:6.2f}% , fee of {:8.3f} sat and a ppm of {:5} ".format(
+                self._probability * 100, self._routing_fee/1000, int(self._routing_fee * 1000 / self._amount))
+        return description
+
+    @property
+    def path(self) -> list[Channel]:
+        """Returns the path of the attempt.
+
+        :return: the list of Channels that the path consists of
+        :rtype: list[Channel]
+        """
+        return self._path
+
+    @property
+    def amount(self) -> int:
+        """Returns the amount of the attempt.
+
+        :return: the amount that was tried to send in this Attempt
+        :rtype: int
+        """
+        return self._amount
+
+    @property
+    def status(self) -> AttemptStatus:
+        """Returns the status of the attempt.
+
+        :return: returns the state of the attempt
+        :rtype: AttemptStatus
+        """
+        return self._status
+
+    @status.setter
+    def status(self, value: AttemptStatus):
+        """Sets the status of the attempt.
+
+        A flag to describe if the path failed, succeeded or was used for settlement (see enum SettlementStatus)
+
+        :param value: Current state of the Attempt
+        :type value: AttemptStatus
+        """
+        if not self._status == value:
+            # remove allocated amounts when Attempt status changes from PLANNED
+            if self._status == AttemptStatus.PLANNED and not value == AttemptStatus.INFLIGHT:
```

Comment:
Why not inflight?

I am wondering if we should have a setter API or if we can only inc status by 1.

That being said I see the idea. While i think it is -given the design of the code- the only reasonable option i think it demonstrates that I messed something up. (It feels really wrong to do the amounts tweeking here)